File: client/src/main/java/client/api/tls/ClientConnectionThread.java
package client.api.tls;

import javax.net.ssl.SSLSession;
import javax.net.ssl.SSLSocket;
import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;

/**
 * Created by prayagupd
 * on 11/14/16.
 */

// Thread handling the socket to server
public class ClientConnectionThread extends Thread {

    private SSLSocket tlsSocket = null;
    private String message = "";

    public ClientConnectionThread(SSLSocket tlsV1Socket, String message) {
        this.tlsSocket = tlsV1Socket;
        this.message = message;
    }

    public void run() {
        tlsSocket.setEnabledCipherSuites(tlsSocket.getSupportedCipherSuites());

        try {
            // Start handshake
            tlsSocket.startHandshake();

            // Get session after the connection is established
            SSLSession tlsV1Session = tlsSocket.getSession();

            System.out.println("[INFO] ClientConnectionThread SSLSession :");
            System.out.println("\tProtocol : " + tlsV1Session.getProtocol());
            System.out.println("\tCipher suite : " + tlsV1Session.getCipherSuite());

            // Start handling application content
            OutputStream socketSendChannel = tlsSocket.getOutputStream();
            PrintWriter httpRequestToSendChannel = new PrintWriter(new OutputStreamWriter(socketSendChannel));
            // Write data
            httpRequestToSendChannel.println(message);
            httpRequestToSendChannel.println();
            httpRequestToSendChannel.flush();

            InputStream receiveChannel = tlsSocket.getInputStream();
            BufferedReader receiveChannelReader = new BufferedReader(new InputStreamReader(receiveChannel));

            String response = null;
            while ((response = receiveChannelReader.readLine()) != null) {
                System.out.println("[INFO] ClientConnectionThread received : " + response);

                if (response.trim().equals("HTTP/1.1 200\r\n")) {
                    break;
                }
            }

            tlsSocket.close();
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }
}

File: client/src/main/kotlin/client/api/tls/ClientConnectionThread.kt
package client.api.tls

import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.io.PrintWriter
import javax.net.ssl.SSLSocket

/**
 * Created by prayagupd
 * on 11/14/16.
 */

// Thread handling the socket to server
class ClientConnectionThread(var tlsV1Socket: SSLSocket, val message: String) : Thread() {

    override fun run() {
        tlsV1Socket.enabledCipherSuites = tlsV1Socket.supportedCipherSuites

        try {
            // Start handshake
            tlsV1Socket.startHandshake()

            // Get session after the connection is established
            val tlsV1Session = tlsV1Socket.session

            println("[INFO] ClientConnectionThread SSLSession :")
            println("\tProtocol : " + tlsV1Session.protocol)
            println("\tCipher suite : " + tlsV1Session.cipherSuite)

            // Start handling application content
            val socketSendChannel = tlsV1Socket.outputStream
            val httpRequestToSendChannel = PrintWriter(OutputStreamWriter(socketSendChannel))
            // Write data
            httpRequestToSendChannel.println(message)
            httpRequestToSendChannel.println()
            httpRequestToSendChannel.flush()

            val receiveChannel = tlsV1Socket.inputStream
            val receiveChannelReader = BufferedReader(InputStreamReader(receiveChannel))

            var response = receiveChannelReader.readLine()
            while (response != null) {
                println("[INFO] ClientConnectionThread received : " + response)

                if (response.trim { it <= ' ' } == "HTTP/1.1 200\r\n") {
                    break
                }
                response = receiveChannelReader.readLine()
            }

            tlsV1Socket.close()
        } catch (ex: Exception) {
            ex.printStackTrace()
        }

    }
}
